projects/storefrontlib/src/cms-components/user/checkout-login/checkout-login.component.ts

```diff
@@ -1,30 +1,81 @@
-import { Component, OnInit } from '@angular/core';
-import { FormGroup, FormBuilder, Validators } from '@angular/forms';
-import { CartService } from '@spartacus/core';
+import { Component } from '@angular/core';
+import { AbstractControl, FormBuilder, Validators } from '@angular/forms';
+import { FormUtils } from '../../../shared/utils/forms/form-utils';
 import { CustomFormValidators } from '../../../shared/utils/validators/custom-form-validators';
+import {
+  CartService,
+  GlobalMessageService,
+  GlobalMessageType,
+} from '@spartacus/core';
 
 @Component({
   selector: 'cx-checkout-login',
   templateUrl: './checkout-login.component.html',
 })
-export class CheckoutLoginComponent implements OnInit {
-  form: FormGroup;
-
-  constructor(private fb: FormBuilder, private cartService: CartService) {}
-
-  ngOnInit(): void {
-    this.form = this.fb.group({
-      userId: ['', [Validators.required, CustomFormValidators.emailValidator]],
-      userIdConf: [
+export class CheckoutLoginComponent {
+  form = this.formBuilder.group(
+    {
+      email: ['', [Validators.required, CustomFormValidators.emailValidator]],
+      emailConfirmation: [
         '',
         [Validators.required, CustomFormValidators.emailValidator],
       ],
-      termsandconditions: [Validators.requiredTrue],
-    });
+      termsAndConditions: ['', [Validators.requiredTrue]],
+    },
+    { validator: this.emailsMatch }
+  );
+  private submitClicked = false;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    protected globalMessageService: GlobalMessageService,
+    protected cartService: CartService
+  ) {}
+
+  isNotValid(formControlName: string): boolean {
+    return FormUtils.isNotValidField(
+      this.form,
+      formControlName,
+      this.submitClicked
+    );
+  }
+
+  isEmailConfirmInvalid(): boolean {
```

Comment:
I have a quick question about this. If I enter an email and never select the confirm email form, click the terms and condition box and then click continue there is no message under the confirm email form. Is this wanted?

Reply:
Nope